pandas/plotting/_core.py

```diff
@@ -862,10 +881,14 @@ def line(self, x=None, y=None, **kwargs):
 
             >>> lines = df.plot.line(x='pig', y='horse')
         """
-        return self(kind="line", x=x, y=y, **kwargs)
 
-    def bar(self, x=None, y=None, **kwargs):
-        """
+    @Appender(_line_docs % _shared_docs)
```

Comment:
Rather than putting this all in Appender you can use Appender with Substitution decorators. Check pandas.core.frame for pre-existing examples

Reply:
Just to be sure: do you mean to replace
```
@Appender(_line_docs % _shared_docs)
```
by 
```
@Substitution(_shared_docs)
@Appender(_line_docs)
```
???

For me this is just more decorators making the code more complex, but not really nicer. Even in pandas.core.frame there are counter examples:
```
@Appender(_shared_docs["align"] % _shared_doc_kwargs)
```
So, do we really want that many decorators?

Reply:
OK sure. Looks like some CI failures can you merge master and we can get this one in